src/metabase/driver/crate/util.clj

```diff
@@ -12,8 +13,16 @@
     :milliseconds (kutils/func (str "TRY_CAST(%s as TIMESTAMP)") [expr])))
 
 (defn- date-trunc [unit expr]
-  "date_trunc('interval', timestamp): truncates a timestamp to a given interval"
-  (k/sqlfn :DATE_TRUNC (kx/literal unit) expr))
+  "date_trunc('interval', timezone, timestamp): truncates a timestamp to a given interval"
+  (if (or (nil? (driver/report-timezone)) (= (count (driver/report-timezone)) 0))
```

Comment:
Accessing `driver/report-timezone` directly is not supported.  We provide access to any global settings information you might need directly on the query when it's passed in for execution, and in the case of the generic-sql driver implementation it creates a binding against `*query*` that you can easily access to get anything out of the query when you need it.

So you should be getting the report-timezone via `(get-in *query* [:settings :report-timezone])`

This is also more convenient because the value is fully normalized, so it will be nil if no timezone value has been set, otherwise it will be the user chosen timezone.